doc/kcollectd.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH KCOLLECTD 1 "9 February 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
kcollectd \- view collectd datacollections
.SH SYNOPSIS
.B kcollectd
.\"RI [ options ] " files" ...
.SH DESCRIPTION
.B Kcollectd 
is a small applications that allows to view 
.B rrd
datacollections that have been created by 
\fBcollectd\fR(1). 
.PP
It is quite limited, but allows to zoom and scroll around and has an
automatic update and follow mode, which makes it some kind of
status-monitor.
.PP
At the left side of the window is a tree, that lists all
\fBcollectd\fR(1) datasources that
.B kcollectd
has found. You can select one or more of them with the mouse.
.PP
At the right the selected datasources are shown as graph. You can move
the graph to the left or right with the mouse und zoom in or out with
the mousewheel or the zoombuttons.
.PP
At the lower left there is a auto-update-button that allows to switch
.B kcollectd 
into the update-and-follow-mode. In this mode the graph still can be
zoomed, but always displays 
.I now
near the right edge and can not be scrolled any more.
.SH AUTHOR
kcollectd was written by M G Berberich.
.PP
This manual page was written by M G Berberich <berberic@fmi.uni-passau.de>.
.SH "SEE ALSO"
.BR collectd(1)
.BR rrdtool(1)


ChangeLog
2010-01-31:
	Version 0.9
	session-management
	
2009-11-30:
	Version 0.8.2
	make it compile with binutils-gold (see debian Bug#554939)

2009-10-11:
	Version 0.8.1
	added a manual to make the help-menu-entry work.

2009-06-06:
	Version 0.8
	added drag-n-drop Pixmap
	make datasource-tree hideable
	multiline legends

2009-06-01:
	read argument from commandline to load file
	load/save config in xml-file

2009-05-31:
	added Menus
	main window now is a KMainWindow

2009-05-26:
	Version 0.7
	context-menu
	changed functionality of gui.
		now datasources has to be dragged to graph-widget

2009-05-10:
	widgets allows multigraphs

2009-05-03:
	Version: 0.6
	added auto-update-and-follow-mode to manpage
	changed from reading command-output of “rrdtool info”
		to using the (new) function in rrd API

2009-04-19:
	Ported to KDE4
	single leave-entries are no longer shown in tree

2009-03-31:
	fixed overlapping of labels on flat graphs
	improved horzontal-lines (there can now be 10, 5 or 2 in a region)

2009-01-08:
	Version 0.3
	fixed bug reported by Daniel Clark (this was a security problem)

2008-08-01:
	removed dependency on kglobalsettings.h from graph-widget
	most of minmax → misc.cc
	code-cleanup
	removed y_range and base vom Graph-widget-class

2008-07-31:
	Version 0.2
	multi-select. The user now can select multible datasources.
	  There might be problems if rrd-layout differs between datasources
	some cleanup

2008-07-23
	some cleanup, time-axes

2008-07-14
	removed the label displaying the datasource
	added last-{hour,day,week,month}-buttons
	added some sensiblity-checks to zooming
	fixed Qt-problem with strftimes localized output
	some more comments
	color_-variables
	KDE-defaultcolor for text
	translation de fixed
	name as a graph-label (but inactive)

2008-07-10
	translation



kcollectd/kcollectd.cc
/*
 * This file is part of the source of kcollectd, a viewer for
 * rrd-databases created by collectd
 * 
 * Copyright (C) 2008 M G Berberich
 *
 * This program is free software: you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <string>
#include <iostream>
#include <exception>

#include <boost/filesystem.hpp>

#include <QTreeWidgetItem>

#include <KAboutData>
#include <KCmdLineArgs>
#include <KApplication>
#include <KMessageBox>
#include <KLocale>

#include "../config.h"

#include "gui.h"

int main(int argc, char **argv)
{
  using namespace boost::filesystem;

  std::vector<std::string> rrds;
  KAboutData about("kcollectd", "kcollectd", 
	ki18n("KCollectd"), VERSION, 
	ki18n("Viewer for Collectd-databases"),
 	KAboutData::License_GPL,
	ki18n("© 2008, 2009 M G Berberich"),
	ki18n("Maintainer and developer"),
	"http://www.forwiss.uni-passau.de/~berberic/Linux/kcollectd.html",
	"berberic@fmi.uni-passau.de");
  about.addAuthor(ki18n("M G Berberich"), KLocalizedString(),
	"M G Berberich <berberic@fmi.uni-passau.de>", 
	"http://www.forwiss.uni-passau.de/~berberic");
  about.setTranslator(ki18nc("NAME OF TRANSLATORS", "Your names"),
	ki18nc("EMAIL OF TRANSLATORS", "Your emails"));
  KCmdLineArgs::init(argc, argv, &about);

  KCmdLineOptions options;
  options.add("+[file]", ki18n("A kcollectd-file to open"));
  KCmdLineArgs::addCmdLineOptions( options );

  KApplication application;
  KCmdLineArgs *args = KCmdLineArgs::parsedArgs();  
  try {
    if (application.isSessionRestored()) {
      kRestoreMainWindows<KCollectdGui>();
    } else {      
      KCollectdGui *gui = new KCollectdGui;
      // handling arguments
      if(args->count() == 1) gui->load(args->arg(0));
      gui->setObjectName("kcollectd#");
      gui->show();
    }
  } 
  catch(basic_filesystem_error<path> &e) {
    KMessageBox::error(0, i18n("Failed to read collectd-structure at \'%1\'\n"
		"Terminating.", QString(RRD_BASEDIR)));
    exit(1);
  }

  return application.exec();
}
